services/edit/queue.js

```diff
@@ -0,0 +1,55 @@
+const Queue = require('promise-queue'),
+  progress = require('../progress'),
+  maxConcurrency = 1,
+  maxQueue = Infinity,
+  queue = new Queue(maxConcurrency, maxQueue),
+  queueCache = {}; // make sure we don't do the same api call over and over if it takes too long
+
+/**
+ * add promise to queue
+ * note: each promise is resolved sequentially,
+ * but promises may contain async children (e.g. Promise.all([a, few, api, calls]))
+ * @param {Function} fn
+ * @param {Array}   args
+ * @param {string} [progressType] 'publish', 'schedule', add this if you need to end with a different progress bar color
+ * @returns {Promise}
+ */
+function add(fn, args, progressType) {
+  const cacheHash = fn.name + JSON.stringify(args);
+
+  let newPromise;
```

Comment:
Why is this a `let` instead of a `var`? I don't _really_ care, but we're not dealing with block scope here, so I just wanted to make sure there wasn't a different reasoning